lib/hydra/pcdm/models/concerns/collection_behavior.rb

```diff
@@ -59,13 +59,15 @@ def collection_ancestor? collections
     end
 
     def ancestor? collection
-      return true if collection == self
+      return true if (( !collection.id.nil? && !self.id.nil? ) && ( collection.id == self.id ))
+      return true if (( collection.id.nil? || self.id.nil? ) && ( collection == self ))
```

Comment:
I did a quick test to confirm that Ruby does the first evaluation of the or first every time.  So the way this is written, the full object comparison will happen every time.  If that fails, then there is a check of class, id, and new_record, which I would expect to always fail if the object comparison failed.